torchsparse/utils/kernel_region.py

```diff
@@ -1,18 +1,32 @@
 import numpy as np
 import torch
+from typing import Union, List, Tuple
 
 __all__ = ['KernelRegion']
 
 
 class KernelRegion:
     def __init__(self,
-                 kernel_size: int = 3,
-                 tensor_stride: int = 1,
-                 dilation: int = 1,
-                 dim=[0, 1, 2]) -> None:
+                 kernel_size: Union[int, List[int], Tuple[int, int, int]] = 3,
+                 tensor_stride: Union[int, List[int], Tuple[int, int, int],
+                                      torch.Tensor] = 1,
+                 dilation: Union[int, List[int], Tuple[int, int, int]] = 1,
+                 dim: List[int] = [0, 1, 2]) -> None:
```

Comment:
I'm not sure if we need to have a class since we only use it to get the kernel offsets.

Reply:
Recently I found that it also makes things easier when we are determining output positions in generalized sparseconv, so I tend to keep that class.